### PR TITLE
CI: docker deployment

### PR DESCRIPTION
**Pull request by eilidhmacnicol**
This PR will automatically deploy a new Docker image when a new NiRodents tag is released

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -53,6 +53,50 @@ jobs:
           command: |
             python -m twine upload dist/*
 
+deploy_docker:
+    machine:
+      image: ubuntu-1604:202007-01
```
Comment (oesteban) [inline .circleci/config.yml:58]: okay, sorry, I'm realizing that mriqc might not be a good reference here. this image is deprecated and will stop working next week :(

File: .circleci/config.yml [hunk 2]
```diff
@@ -168,9 +212,19 @@ workflows:
...
+      - deploy_pypi:
+          requires:
+            - test_package
+          filters:
+            branches:
+              ignore: /.*/
+            tags:
+              only: /.*/
+
       - deploy_package:
```
Comment (oesteban) [inline .circleci/config.yml:224]: I think you meant
```suggestion
      - deploy_docker:
```

File: .circleci/config.yml (hunk 1 above)
Comment (oesteban) [reply .circleci/config.yml]: yes - that seems to be the easiest fix https://circleci.com/blog/ubuntu-14-16-image-deprecation/

**Comment by eilidhmacnicol** [comment]
Okay - if there's nothing else you can spot, @oesteban, I'll merge this tomorrow PM (travelling in the AM), cut a new release, and then continue with nipreps/mriqc#969

File: .circleci/config.yml [hunk 3]
```diff
@@ -168,9 +212,19 @@ workflows:
...
+          filters:
+            branches:
+              ignore: /.*/
+            tags:
+              only: /.*/
+
+      - deploy_docker:
           requires:
             - test_package
+            - deploy_pypi
```
Comment (oesteban) [inline .circleci/config.yml:227]: You want to make sure your image is built.

```suggestion
            - build
            - deploy_pypi
```

**Review by oesteban — COMMENTED** [review]
I'm not positive a docker image needs to be pushed, tbh

**Comment by eilidhmacnicol** [comment]
What are the alternatives for updating DockerHub?

**Comment by oesteban** [comment]
The question is, why do you need for an image of nirodents to exist at all?

**Comment by eilidhmacnicol** [comment]
We've had a lot of interest in artsBrainExtraction on neurostars and through email. I know NiBrex is the plan going forward, but until that's ready to go we're potentially losing users who try an outdated version of the image on DockerHub which I don't expect will give results that are as good as the latest release.

**Comment by oesteban** [comment]
Sure, what I can't see is how this is a holdup for nipreps/mriqc#969

**Comment by eilidhmacnicol** [comment]
Oh, I see! It isn't necessary for mriqc but I thought that it could be included before I cut a new release

**Pull request merged**